server/data/data-cassandra/src/main/java/org/apache/james/domainlist/cassandra/CassandraDomainList.java

```diff
@@ -100,8 +89,8 @@ protected boolean containsDomainInternal(Domain domain) throws DomainListExcepti
     @Override
     public void addDomain(Domain domain) throws DomainListException {
         boolean executed = executor.executeReturnApplied(insertStatement.bind()
-            .setString(DOMAIN, domain.asString()))
-            .block();
+                .setString(DOMAIN, domain.asString()))
+            .blockOptional().orElseThrow();
```

Comment:
Why not `.block()` ?

Reply:
two way is the same, 
`block()` can be throw NullPointerException from Reactor (warning from IDE)
`blockOptional().orElseThrow()` for assert that